### DISCUSSION TITLE
Display TW as JSON with its own URI?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5525] [Q&A]
For integration with other software it would be useful if a TW could be directly viewed in JSON format. Is this somehow possible to type some magic URL and have the TW presented as JSON?

Perhaps `...mytiddlywiki.html?=JSON`

I am aware that one can *export* to JSON format but that of course creates a separate file.

Thanks.

<hr>

Little nod to @Marxsal and also Mr. JSON himself @joshuafontany -  but the question is to everyone.

**Comment by pmario** [comment]
You want to see the "store" or the whole wiki, including the html-elements as JSON. IMO the HTML stuff wouldn't make much sense for most users

**Reply by twMat** [reply]
>the HTML stuff wouldn't make much sense

Very good point! For integration with other software I agree the *content* is likely all that one is after.

By the way, would it be correct to call what I'm after a "API"?

And, is there currently any existing API in TW?

**Comment by saqimtiaz** [comment]
If the JSON is meant to be consumed by other serices: With Node.js, yes this is possible.

With single file the only way you could get it to work is by saving a JSON file with every single file TW. You need to provide just the JSON data in the file, so tricks like using a query in the URL to display just the JSON data would not be enough.

Note that with a separate JSON file this would still run into CORS restrictions and JSONP would only be an option with hard coded callbacks, and in general JSONP is no longer recommended due to security considerations.

**Reply by twMat** [reply]
Thanx @saqimtiaz

**Comment by saqimtiaz** [comment]
TiddlyHost now offers JSON end points for the wikis and also has CORS enabled.

See https://github.com/simonbaird/tiddlyhost/issues/112#issuecomment-813700504
